### ISSUE TITLE
A false positive about the rule DM_EXIT

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1811]
**SpotBugs Version:** 4.4.2
**Rule:** [DM_EXIT](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-method-invokes-system-exit-dm-exit)

Hi! I think I found a false positive about the rule DM_EXIT.
**Minimal Code Sample**
```java
public static final boolean tag;
static {
    tag = false;
}
public void func() {
    if(tag) {
        System.exit(0);  // A false violation warning reported here
    }
}
```
SpotBugs doesn't report a warning about the below case, so I think this problem is caused by the ignorance of final variables initialization.
```java
public static final boolean tag = false;
public void func() {
    if(tag) {
        System.exit(0);
    }
}
```
**Expected outcome:**
SpotBugs shouldn't report a violation at line 7 of the first code sample, but it does. This is a false positive.

**Running SpotBugs through:** [CLI]

**Comment by Zustin** [comment]
I found the rule [DMI_EMPTY_DB_PASSWORD](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-empty-database-password-dmi-empty-db-password) has similar proble, the code example is below:
```java
public static final String s;
static {
    s = "";
}
Connection getConnection1() throws SQLException {
    return DriverManager.getConnection("jdbc:hsqldb:mem:test", "sa", s); // to submit
}
```
